### ISSUE TITLE
Can we drop rust? (And maybe llvm?)

### ISSUE THREAD

**Issue by apjanke** [octave-app/octave-app#248]
The build has a 970 MB rust installation, which takes quite a while to build too. I'm not even sure what pulls that in; maybe svn? Would be nice to skip it entirely.

It should be pruned at app build time, but I see it sticking around in the releases, too. Fix that, at least.

Maybe it's librsvg, libmpc, or gdk-pixbuf?

Goes with https://github.com/octave-app/octave-app/issues/247.

## TODO

* [X] Get rust pruned properly from the build.
* [nope] See if we can skip it in the first place.
  * [X] Find out what deps (direct and indirect) are actually pulling rust in. (answer: it's librsvg)
* [testing] Prune llvm from the build.

## References

* Related:
  * https://github.com/octave-app/octave-app/issues/247

**Comment by apjanke** [comment]
Heck, maybe we could ditch llvm, too? I think maybe only gcc is used for compiling oct/mex files, so maybe llvm is an Octave.app-build-time-only dependency, and could be pruned, or maybe avoided in the first place? Dunno.

<img width="957" alt="image" src="https://github.com/octave-app/octave-app/assets/2618447/c3120ebe-3459-4034-b5ae-d9a808256873">

**Comment by apjanke** [comment]
Heh. Looks like the only thing using llvm was rust.

```
[Formula] $ brew uses --installed --include-build llvm
rust
[Formula] $ brew uses --installed llvm
rust
[Formula] $
```

After removing `rust` from the Octave.app build, `brew uses ... llvm` outputs nothing. Think I can prune it, and seriously cut the installation size.

**Comment by apjanke** [comment]
Aw, rats. Even with removing the subversion dependency, it still pulls in rust and llvm. librsvg has a `depends_on "rust" => :build` dependency.

Oh well. It's still nice to have all that stuff in the main build step instead of a separate `brew install subversion` before the main build. And I'm more confident that I can prune both rust and llvm after the build now, and get the install size way down. Added llvm to the list of build-time-only packages to prune at the end of the build.

**Comment by apjanke** [comment]
Pruning rust and llvm from the build did some good.

* Before: 6 GB app, 1.76 GB DMG (alpha2)
* After: 3 GB app, 1.05 GB DMG (alpha3)

No additional failures in the test suite.

```
Summary:

  PASS                            17672
  FAIL                                3
  REGRESSION                          1
  XFAIL (reported bug)               60
  SKIP (missing feature)              3
  SKIP (run-time condition)          46
```

Got timing info now too. Just over two hours for the build on my M1 Max MBP. That's nice in case I forget to slap a `time` on it.

```
DMG ready at: ./build/Octave-8.4.0_alpha3.dmg
elapsed time: action: 01:25
elapsed time: whole build: 141:52
Build finished.
```

Calling this good and closing as Fixed-Enough. Unrealistic to eliminate the build-time dependency at this point, I think.
